examples/platforms/utils/soft_source_match_table.h

```diff
@@ -37,20 +37,31 @@
 #define SOFT_SOURCE_MATCH_TABLE_H
 
 #include "openthread-core-config.h"
+#include <openthread/platform/radio.h>
+
+#include <stdint.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#if defined(RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM) || defined(RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM)
+#if !defined(RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM)
+#define RADIO_CONFIG_SRC_MATCH_SHORT_ENTRY_NUM 0
+#endif
+
+#if !defined(RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM)
+#define RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM 0
+#endif
```

Comment:
If the macro is not defined. The macro is set to 0 in `#if ...` statement. So the code between line 48 and 54 can be removed.

Reply:
Yes, I left this code to avoid a warning when compiling with -Wundef.

Reply:
nit: how about `#ifndef` rather than `#if !defined()`?

Reply:
Changed.